Sources/AsyncHTTPClient/HTTPDecompressor.swift

```diff
@@ -0,0 +1,151 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2018-2019 Swift Server Working Group and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import CNIOExtrasZlib
+import NIO
+import NIOHTTP1
+
+private enum CompressionAlgorithm: String {
+    case gzip
+    case deflate
+}
+
+extension z_stream {
```

Comment:
@weissi this code "violates" the [sswg  min requirements](https://github.com/swift-server/sswg/blob/master/process/incubation.md#minimal-requirements) due to the use of unsafe pointers

Reply:
maybe the right choice here is to extract the compression logic to its own library, so its reusable and we concentrate the unsafe into one place

Reply:
FWIW, I don't think it violates the min requirements, as it is covered by the "unless interfacing with C" exemption.

Reply:
That said, we can and should write the interface in _this_ module to avoid exposing unsafe constructs.